### PR TITLE
Loader for buttons G3 2024 v4 #14648

### PR DESCRIPTION
**Pull request by a22maxra**
Buttons now have loading capabilities when waiting for a request to be fulfilled, replace old buttons/inputs
Old:
```
<input id='saveRepo' class='submit-button' style='float:right; margin-right:5px;' type='button' value='Save' title='Save Changes' onclick='validateForm("githubPopupWindow");' />
```
New, make a container and include `loadingButton.js`
``` js
<script src="../Shared/loadingButton.js"></script>

<div id="buttonContainerSaveRepo"></div>

```

Within `loaderButton.js` fill in the function to be called, the argument, label and containerID in the _actions_ object
```js
    const actions = {
      saveRepo: {
        action: validateForm, // Function
        argument: "githubPopupWindow", // Argument to pass to the function
        buttonLabel: "Save Repo", // Button label
        containerId: "buttonContainerSaveRepo" // Container ID
      },
```

When adding more buttons, create a new container, make sure `loaderButton.js` is included in the php file and add the correct arguments to the _actions_ object.

For new courses you need to fill in the information and give it a version for the button to work, i don't know why this is, the `onClick` should work either way from my understanding but this should not be a problem since you would fill these fields out before getting data from Github in a real world scenario.

This solution allows buttons to be changed to the loader version incrementally and if something is not working the old versions of the inputs/buttons can be used in the mean time. If the solution is satisfactory more functionality can be added and more buttons can be switched out, right now the button to edit information about a course and the get Github data are loading buttons. 

image

### CODE REVIEW

File: DuggaSys/courseed.php [hunk 1]
```diff
@@ -141,7 +142,7 @@
 				</div>
     		</div>
     		<div style='float:right; padding-top:20px; width: 464px;' >
-    			<input id='saveCourse' class='submit-button' type='button' value='Save' title='Save changes' onclick="validateForm('editCourse')" />
+				<div id="buttonContainerSaveCourse"></div>
     		</div>
```
Comment (a22willi) [inline DuggaSys/courseed.php:146]: Indentation.

**Review by a22willi — CHANGES_REQUESTED** [review]
Could you make the spinner slightly smaller, and have the button appear disabled when clicked loading? As detailed in the original issue.

File: DuggaSys/courseed.php (hunk 1 above)
Comment (a22maxra) [reply DuggaSys/courseed.php:146]: Looks good in VSCode

**Review by a22willi — APPROVED** [review]
Looks good.

**Review by a22willi — APPROVED** [review]
Looks good.

**Pull request merged**